lib/possible_bipartition.rb

```diff
@@ -1,4 +1,36 @@
 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+  return true if dislikes.empty? || dislikes.length < 2
+
+  # group_hash includes visited & queued nodes, key is the node & value is the group
+  # group can be 1 or 2
+  group_hash = {}
+  q = []
+
+  # add first, connected node to q & group_hash
+  count = 0
+  i = 0
+  until count == 1 && i < dislikes.length
```

Comment:
What if the graph is unconnected like:

```ruby
adjacency_list = {
  1 => [2, 3],
  2 => [1, 3],
  3 => [1, 2],
  4 => [5, 6],
  5 => [4, 6],
  6 => [4, 5]
}
```

In this graph there are two unconnected groups of nodes.  I'll update my tests to catch this scenario.